test/shelf_pubserver_test.dart

```diff
@@ -402,17 +407,18 @@ main() {
           var newUrl = getUri('/api/packages/versions/new');
           var uploadUrl = getUri('/api/packages/versions/newUpload');
           var finishUrl = getUri('/api/packages/versions/newUploadFinish');
-          var mock = new RepositoryMock(supportsUpload: true,
+          var mock = new RepositoryMock(
+              supportsUpload: true,
               uploadFun: (Stream<List<int>> stream) {
-            return stream.fold([], (b, d) => b..addAll(d)).then((List<int> d) {
-              expect(d, equals(tarballBytes));
-              return new PackageVersion('foobar', '0.1.0', '');
-            });
-          });
+                return stream.fold([], (b, d) => b..addAll(d)).then((d) {
```

Comment:
Instead of removing the type, you could add it at even more places, e.g. `<int>[]`

Reply:
Yeah, although inference should help here.

Turns out the type isn't needed – for the body, so removing things makes the new fuzzy arrow work.

Reply:
Wondering if this is in fact true, whether strong mode inference will make `[]` a `<int>[]`.